coders/composio_coders/swe.py

```diff
@@ -0,0 +1,227 @@
+        self.repo_name = args.repo_name
+        if not self.issue_config.issue_id:
+            raise ValueError("no git-issue configuration is found")
+
+        # initialize composio toolset
+        tool_set = ComposioToolSet()
+        self.composio_toolset = tool_set.get_tools(apps=[App.LOCALWORKSPACE,
+                                                         App.CMDMANAGERTOOL,
+                                                         App.HISTORYKEEPER,])
+        # initialize agent-related different prompts
+        self.agent_role = "You are the best programmer. You think carefully and step by step take action."
+        self.agent_goal = "Help fix the given issue / bug in the code. And make sure you get it working."
+        self.expected_output = "A patch should be generated which fixes the given issue"
+        self.agent_backstory_tmpl = args.agent_backstory_tmpl
+        self.issue_description_tmpl = args.issue_description_tmpl
+        # initialize logger
+        self.logger = logger
+        # initialize agent logs and history dict
+        self.task_output_logs = ""
+        self.agent_logs = {}
+        self.agent_history = {}
+        self.current_logs = []
+        self.setup_logs_path(args.agent_output_dir)
+
+    def setup_logs_path(self, agent_output_dir_path):
+
+        task_output_dir = script_dir / Path(agent_output_dir_path) / Path(
+            LOGS_DIR_NAME_PREFIX + "_" + datetime.datetime.now().strftime("%Y-%m-%d_%H-%M-%S"))
+        self.task_output_logs = task_output_dir / Path(AGENT_LOGS_JSON_PATH)
+        if not os.path.exists(task_output_dir):
+            os.makedirs(task_output_dir)
+
+    def save_history(self, instance_id):
+        self.agent_logs[instance_id] = self.current_logs
+        with open(self.task_output_logs, "w") as f:
+            f.write(json.dumps(self.agent_logs))
+
+    def add_in_logs(self, step_output):
+        if isinstance(step_output, langchain_core.agents.AgentFinish):
+            self.current_logs.append({
+                "agent_action": "agent_finish",
+                "agent_output": step_output.return_values,
+            })
+        if isinstance(step_output, list) and step_output:
+            agent_action_with_tool_out = step_output[0]
+            if isinstance(agent_action_with_tool_out[0], langchain_core.agents.AgentAction):
+                agent_action = agent_action_with_tool_out[0]
+                tool_out = agent_action_with_tool_out[1] if len(agent_action_with_tool_out) > 1 else None
+                self.current_logs.append({"agent_action": agent_action.json(), "tool_output": tool_out})
+            else:
+                self.logger.info("type of step_output: %s", type(agent_action_with_tool_out[0]))
+        else:
+            self.logger.info("type is not list: %s", type(step_output))
+
+    def get_llm(self):
+        # get model_env config from path 
+        try:
+            model_env_path = config_dir / Path(MODEL_ENV_CONFIG_PATH)
```

Comment:
This is not consistent with the `setup` command, on the `setup` we're storing the model config in the current working directory (https://github.com/ComposioHQ/composio/pull/163/files#diff-97f4277ad1ff3c6363f41502ff3e2b66a561a755d77ad62ddca72cbe8b6035a8R43) and here's we're using `script_dir` as base for loading the model config.

Reply:
fixed